### DISCUSSION TITLE
Best practices for sending push notifications with AdonisJS

### DISCUSSION THREAD

**Discussion by Anthony-B404** [adonisjs/core#4761] [Help]
Hi everyone,

I'm currently working on an application with an AdonisJS backend and PWA built with Vue.js. I’m looking to integrate push notifications into the app.

I would love to get some advice or insights from the community on the best practices for sending push notifications from an AdonisJS backend. Specifically, I’m wondering:

Is it easy to integrate services like Firebase Cloud Messaging (FCM), OneSignal, or any other popular push notification service with AdonisJS?
Are there any open-source modules in the AdonisJS ecosystem that make this integration smoother?
What are the best practices to ensure secure and scalable push notifications in an AdonisJS environment?
Any suggestions or experience shared would be greatly appreciated!

Thank you in advance for your help and advice.

**Comment by MaximeMRF** [comment]
Hi @Anthony-B404,

You can send push notifications using [Transmit](https://docs.adonisjs.com/guides/digging-deeper/transmit#transmit)
